fvm/transactionInvoker.go

```diff
@@ -521,18 +527,36 @@ func (executor *transactionExecutor) checkDependencies() error {
 			dependencies = append(dependencies, addressLoc)
 		}
 	}
+	if len(dependencies) == 0 {
+		// this is an unlikely case, but we don't need to check dependencies
+		// if there are none
+		return nil
+	}
+	// sort the dependencies so the execution order is deterministic
+	slices.SortFunc(
+		dependencies,
+		func(a common.AddressLocation, b common.AddressLocation) int {
+			return strings.Compare(a.ID(), b.ID())
+		})
 
 	authorizerSet := make(map[flow.Address]struct{}, len(executor.proc.Transaction.Authorizers))
 	for _, authorizer := range executor.proc.Transaction.Authorizers {
 		authorizerSet[authorizer] = struct{}{}
 	}
 
+	// add the payer as well
 	authorizerSet[executor.proc.Transaction.Payer] = struct{}{}
 
 	auths := make([]flow.Address, 0, len(authorizerSet))
 	for auth := range authorizerSet {
```

Comment:
In the Cadence repo we have a custom linter to warn about such cases (https://github.com/onflow/cadence/tree/master/tools/maprange), we might want to consider adding this to flow-go, too.

Reply:
https://github.com/onflow/flow-go/issues/5967